internal/exp/metrics/metrics.go

```diff
@@ -0,0 +1,123 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package metrics // import "github.com/open-telemetry/opentelemetry-collector-contrib/internal/exp/metrics"
+
+import (
+	"go.opentelemetry.io/collector/pdata/pmetric"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/exp/metrics/identity"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/exp/metrics/streams"
+)
+
+// Merge will merge the metrics data in mdB into mdA, then return mdA.
+// mdB will not be modified. The function will attempt to merge the data in mdB into
+// existing ResourceMetrics / ScopeMetrics / Metrics in mdA if possible. If they don't
+// exist, new entries will be created as needed.
+//
+// NOTE: Any "unnecessary" duplicate entries in mdA will *not* be combined. For example if
```

Comment:
Could you add test case to cover this situation, `Any "unnecessary" duplicate entries in mdA will *not* be combined.`?

Reply:
Done